src/pages/Download.tsx

```diff
@@ -0,0 +1,201 @@
+
+  const isDownloadDisabled = !carCid.trim() || loading;
+
+  return (
+    <>
+      <div className="bg-white rounded-lg shadow p-6 mb-4">
+        <h2 className="text-xl font-bold mb-4">Content retrieval</h2>
+        <div className="mb-4">
+          <label htmlFor="car-id" className="block text-sm font-medium text-gray-700 mb-1">
+            Payload CID
+          </label>
+          <input
+            id="car-id"
+            type="text"
+            className="w-full p-2 border rounded focus:ring-blue-500 focus:border-blue-500"
+            value={carCid}
+            onChange={(e) => setCarId(e.target.value)}
+          />
+        </div>
+
+        <div className="mb-4">
+          <label
+            htmlFor="provider-multiaddr-id"
+            className="block text-sm font-medium text-gray-700 mb-1"
+          >
+            Provider
+          </label>
+          <input
+            id="provider-multiaddr-id"
+            type="text"
+            className="w-full p-2 border rounded focus:ring-blue-500 focus:border-blue-500"
+            value={providerMultiaddr}
+            onChange={(e) => setProviderMultiaddr(e.target.value)}
+          />
+        </div>
+
+        <div className="mb-4">
+          <div className="flex items-center">
+            <input
+              id="extract-car"
+              type="checkbox"
+              className="h-4 w-4 text-blue-600 focus:ring-blue-500 border-gray-300 rounded"
+              checked={shouldExtract}
+              onChange={(e) => setShouldExtract(e.target.checked)}
+            />
+            <label htmlFor="extract-car" className="ml-2 block text-sm text-gray-700">
+              Extract
+            </label>
+          </div>
+        </div>
+
+        <div className="mt-6">
+          <DownloadButton
+            onClick={downloadCar}
+            disabled={isDownloadDisabled}
+            text={loading ? "Downloading..." : "Download"}
+          />
+        </div>
+      </div>
+      <Toaster position="bottom-right" reverseOrder={true} />
+    </>
+  );
+}
+
+async function retrieveContent(payloadCid: CID, provider: Multiaddr, extractContents: boolean = true): Promise<{ title: string, contents: Blob }> {
+  // enable verbose logging in browser console to view debug logs
+  enable("ui*,libp2p*,-libp2p:connection-manager*,helia*,helia*:trace,-*:trace");
+
+  // Create a libp2p node
+  const libp2p = await createLibp2p({
+    transports: [webSockets()],
+    connectionEncrypters: [noise()],
+    streamMuxers: [yamux()],
+    services: {
+      identify: identify(),
+    },
+    // We are currently disabling the connection monitor because it uses ping
+    // protocol under the hood. The storage-provider-server is not listening for pings.
+    connectionMonitor: {
+      enabled: false,
+    },
+  });
+
+  // Create Helia wrapper around the libp2p node
+  const helia = await createHelia({
+    libp2p,
+    blockBrokers: [bitswap()],
+    routers: [libp2pRouting(libp2p)],
+  });
```

Comment:
```suggestion
  enable("ui*,libp2p*,-libp2p:connection-manager*,helia*,helia*:trace,-*:trace");

  // Create a libp2p node
  const libp2p = await createLibp2p({
    transports: [webSockets()],
    connectionEncrypters: [noise()],
    streamMuxers: [yamux()],
    services: {
      identify: identify(),
    },
    // We are currently disabling the connection monitor because it uses ping
    // protocol under the hood. The storage-provider-server is not listening for pings.
    connectionMonitor: {
      enabled: false,
    },
  });

  // Create Helia wrapper around the libp2p node
  const helia = await createHelia({
    libp2p,
    blockBrokers: [bitswap()],
    routers: [libp2pRouting(libp2p)],
  });
```

I suppose this whole config should be done once per application.

Reply:
I think it's fine for now. The node is configured per retrieval.